src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -76,7 +76,7 @@ public void SetContext(EvaluationContext context)
         public FeatureClient(string name, string version, ILogger logger = null, EvaluationContext context = null)
         {
             this._metadata = new ClientMetadata(name, version);
-            this._logger = logger ?? new Logger<Api>(new NullLoggerFactory());
+            this._logger = logger ?? NullLogger<FeatureClient>.Instance;

```

Comment:
Note the logger name/category change here. It doesn't actually matter for the no-op logger, but prestaging the change while I'm at it because this _will_ matter in the post-#181 world because when #181 registers `IFeatureClient` it injects a named logger similar to how `HttpClient` loggers work so that consumer apps can exercise granular control over logging sources.

https://github.com/open-feature/dotnet-sdk/blob/9821a5b1f9be0cc2fce30481d8572449301666fb/src/OpenFeature.Extensions.Hosting/OpenFeatureBuilderExtensions.cs#L109-L113

Such that a consumer app could do something like this in their `appsettings.json` to vary the minimum log levels by provider:

```json
{
  "Logging": {
    "LogLevel": {
      "Default": "Debug",
      "OpenFeature": "Information",
      "OpenFeature.FeatureClient": "Warning",
      "OpenFeature.FeatureClient.No-op Provider": "Warning",
      "OpenFeature.FeatureClient.flagd Provider": "Information",
    }
  }
}
```